### PR TITLE
Handle nans in data

### PR DESCRIPTION
**Pull request by henrikjacobsenfys**
Closing #119 
Closing #118

### CODE REVIEW

File: tests/unit/easydynamics/experiment/test_experiment.py [hunk 1]
```diff
@@ -379,6 +396,53 @@ def test_convert_to_bin_centers(self, experiment):
...
+            experiment_with_data.data.variances[Q_index] ** 0.5,
+        )
+
+    def test_extract_x_y_weights_only_finite_zero_variances(self, experiment_with_data):
+        "Test that _extract_x_y_weights_only_finite raises ValueError when variances contain zeros"
+        # WHEN
+        Q_index = 0
+        invalid_data = experiment_with_data._data.copy()
+        invalid_data.data.variances[Q_index] = 0  # Set variances to zero
+
```
Comment (rozyczko) [inline tests/unit/easydynamics/experiment/test_experiment.py:423]: This sets all variances for Q_index=0 to zero. It means they all pass the `isfinite` check but then fail the `e == 0` check. This is correct, but it would be stronger to also test the case where only some variances are zero while others have NaN/Inf. This would verify that the filtering happens first and the zero-check second.

**Review by rozyczko — APPROVED** [review]
Looks very good! Just a note about the possibility of adding more unit test coverage.

File: tests/unit/easydynamics/experiment/test_experiment.py (hunk 1 above)
Comment (henrikjacobsenfys) [reply tests/unit/easydynamics/experiment/test_experiment.py:423]: Thanks, made one of the variances NaN :)
